Tasks/Common/nuget-task-common/NuGetToolRunner2.ts

```diff
@@ -41,32 +47,47 @@ function prepareNuGetExeEnvironment(
             }
         }
 
-        if (e.toUpperCase() === "NUGET_CREDENTIALPROVIDERS_PATH") {
-            originalCredProviderPath = input[e];
+        // New credential provider
+        if (e.toUpperCase() === "NUGET_PLUGIN_PATHS") {
+            envVarCredProviderPathV2 = input[e];
+            continue;
+        }
 
-            // will re-set this variable below
+        // Old credential provider
+        if (e.toUpperCase() === "NUGET_CREDENTIALPROVIDERS_PATH") {
+            envVarCredProviderPathV1 = input[e];
             continue;
         }
 
         env[e] = input[e];
     }
 
-    let credProviderPath = settings.credProviderFolder || originalCredProviderPath;
-    if (settings.credProviderFolder && originalCredProviderPath) {
-        credProviderPath = settings.credProviderFolder + ";" + originalCredProviderPath;
-    }
-
-    if (authInfo && authInfo.internalAuthInfo){
+    if (authInfo && authInfo.internalAuthInfo) {
         env["VSS_NUGET_ACCESSTOKEN"] = authInfo.internalAuthInfo.accessToken;
         env["VSS_NUGET_URI_PREFIXES"] = authInfo.internalAuthInfo.uriPrefixes.join(";");
     }
 
     env["NUGET_CREDENTIAL_PROVIDER_OVERRIDE_DEFAULT"] = "true";
 
-    if (credProviderPath) {
-        tl.debug(`credProviderPath = ${credProviderPath}`);
-        env["NUGET_CREDENTIALPROVIDERS_PATH"] = credProviderPath;
+    let credProviderPath = null;
+    // If NUGET_CREDENTIALPROVIDERS_PATH is provided
+    // use old cred provider regardles of which version on NuGet
+    if (settings.credProviderFolder != null || envVarCredProviderPathV1 != null) {
+        credProviderPath = buildCredProviderPath(settings.credProviderFolder, envVarCredProviderPathV1);
+
+        if (credProviderPath) {
+            env["NUGET_CREDENTIALPROVIDERS_PATH"] = credProviderPath;
+            tl.debug(`Using V1 credential provider`);
+        }
+    } else {
```

Comment:
This removes the possibility of setting both environment variables at the same time, which a caller may want to do. Let's set these variables independently and leave the responsibility of choosing which ones to the caller. So here the logic should remain unchanged, just add the logic for the v2plugin env var.